widgetssdk/src/main/java/com/glia/widgets/filepreview/data/source/local/InAppBitmapCache.java

```diff
@@ -2,28 +2,21 @@
 
 import android.graphics.Bitmap;
 
-import java.util.HashMap;
-import java.util.Map;
+import androidx.collection.ArrayMap;
 
 public class InAppBitmapCache {
 
-    private static InAppBitmapCache instance;
-
     private InAppBitmapCache() {
     }
 
-    private final Map<String, Bitmap> bitmapsById = new HashMap<>();
+    private final ArrayMap<String, Bitmap> bitmapsById = new ArrayMap<>();
 
     public static InAppBitmapCache getInstance() {
-        if (instance == null) {
-            synchronized (InAppBitmapCache.class) {
-                if (instance == null) {
-                    instance = new InAppBitmapCache();
-                }
-            }
-        }
-
-        return instance;
+        return InstanceHolder.instance;
+    }
+
+    private static final class InstanceHolder {
+        private static final InAppBitmapCache instance = new InAppBitmapCache();
     }
```

Comment:
Could you please explain the benefit of wrapping `InAppBitmapCache` with a class?

Reply:
When the singleton class is loaded, SingletonHelper class is not loaded into memory and only when someone calls the getInstance() method, this class gets loaded and creates the singleton class instance. This is the most widely used approach for the singleton class as it doesn’t require synchronization.

Reply:
Okay, thank you for the explanation.